### PR TITLE
typ{o,e} fixes

### PR DESCRIPTION
**Pull request by ryan-williams**
- First arg to `TileDBCloudError` is supposed to be `http_status: Optional[int]`, but these callsites were passing a `str.
  - I guessed that those should be `body` (and `body` should be allowed to be `str | bytes`, not just `bytes`)
- Couple docstr/comment typos

### CODE REVIEW

**Review by sgillies — CHANGES_REQUESTED** [review]
@ryan-williams Unless I've overlooked one, all these instances of `raise tce.TileDBCloudError` happen when validating inputs before taking the step of creating a graph or node. They are not associated with making a REST server request, right?

Thus, I believe we shouldn't be raising this particular exception in these cases. Let's raise ValueError instead, like at https://github.com/TileDB-Inc/TileDB-Cloud-Py/blob/8c2da9916f1bc5957ea880c04cc31428d7008695/src/tiledb/cloud/dag/dag.py#L2016. ValueError takes just a string.

**Comment by ryan-williams** [comment]
Good point, converted them to `ValueError`s

File: src/tiledb/cloud/dag/dag.py [hunk 1]
```diff
@@ -1083,12 +1078,12 @@ def submit_udf_stage(
             **kwargs,
         )
 
-    def submit_sql(self, *args: Any, **kwargs: Any) -> Node:
+    def submit_sql(self, sql: str, *args: Any, **kwargs: Any) -> Node:
```
Comment (sgillies) [inline src/tiledb/cloud/dag/dag.py:1081]: @ryan-williams If we pick `sql` out of `*args`, then I think we have to pass it explicitly on line 1092, yes?

**Review by sgillies — APPROVED** [review]
@ryan-williams Great! I believe I see one required change.

File: src/tiledb/cloud/dag/dag.py (hunk 1 above)
Comment (ryan-williams) [reply src/tiledb/cloud/dag/dag.py:1081]: yikes, yes, good catch. Looks like it failed some tests as well, thankfully.

Comment (sgillies) [reply src/tiledb/cloud/dag/dag.py:1081]: I'm glad we have a test that covers this! https://github.com/TileDB-Inc/TileDB-Cloud-Py/actions/runs/11842972791/job/33002846291?pr=680#step:12:411

**Comment by ryan-williams** [comment]
Looks like `ConnectToExistingTest::test_running_graph` is failing on `main`, at least some of the time. [Example](https://github.com/TileDB-Inc/TileDB-Cloud-Py/actions/runs/11829336769/job/32960978779#step:12:263):
```
FAILED tests/taskgraphs/test_server_side.py::ConnectToExistingTest::test_running_graph - concurrent.futures._base.TimeoutError: timed out after 120 sec
```

**Comment by sgillies** [comment]
@ryan-williams I'm confident that the failures are unrelated and will merge.

**Pull request merged**